src/test/kotlin/org/rust/lang/core/resolve2/RsDefMapUpdateChangeSingleFileTest.kt

```diff
@@ -0,0 +1,411 @@
+        fn bar() {}
+        fn foo() {}
+    """)
+
+    fun `test swap functions`() = doTestNotChanged("""
+        fn foo() {}
+        fn bar() {}
+    """, """
+        fn bar() {}
+        fn foo() {}
+    """)
+
+    fun `test change item visibility 1`() = doTestChanged("""
+        fn foo() {}
+    """, """
+        pub fn foo() {}
+    """)
+
+    fun `test change item visibility 2`() = doTestChanged("""
+        pub fn foo() {}
+    """, """
+        fn foo() {}
+    """)
+
+    fun `test change item visibility 3`() = doTestChanged("""
+        pub fn foo() {}
+    """, """
+        pub(crate) fn foo() {}
+    """)
+
+    fun `test change item visibility 4`() = doTestNotChanged("""
+        pub(crate) fn foo() {}
+    """, """
+        pub(in crate) fn foo() {}
+    """)
+
+    fun `test change item visibility 5`() = doTestNotChanged("""
+        fn foo() {}
+    """, """
+        pub(self) fn foo() {}
+    """)
+
+    fun `test add item with same name in different namespace`() = doTestChanged("""
+        fn foo() {}
+    """, """
+        fn foo() {}
+        mod foo {}
+    """)
+
+    fun `test remove item with same name in different namespace`() = doTestChanged("""
+        fn foo() {}
+        mod foo {}
+    """, """
+        fn foo() {}
+    """)
+
+    fun `test change import 1`() = doTestChanged("""
+        use aaa::bbb;
+    """, """
+        use aaa::ccc;
+    """)
+
+    fun `test change import 2`() = doTestChanged("""
+        use aaa::{bbb, ccc};
+    """, """
+        use aaa::{bbb, ddd};
+    """)
+
+    fun `test swap imports`() = doTestNotChanged("""
+        use aaa::bbb;
+        use aaa::ccc;
+    """, """
+        use aaa::ccc;
+        use aaa::bbb;
+    """)
+
+    fun `test swap paths in use group`() = doTestNotChanged("""
+        use aaa::{bbb, ccc};
+    """, """
+        use aaa::{ccc, bbb};
+    """)
+
+    fun `test change import visibility`() = doTestChanged("""
+        use aaa::bbb;
+    """, """
+        pub use aaa::bbb;
+    """)
+
+    fun `test change extern crate 1`() = doTestChanged("""
+        extern crate foo;
+    """, """
+        extern crate bar;
+    """)
+
+    fun `test change extern crate 2`() = doTestChanged("""
+        extern crate foo;
+    """, """
+        extern crate foo as bar;
+    """)
+
+    fun `test add macro_use to extern crate`() = doTestChanged("""
+        extern crate foo;
+    """, """
+        #[macro_use]
+        extern crate foo;
+    """)
+
+    fun `test remove macro_use from extern crate`() = doTestChanged("""
+        #[macro_use]
+        extern crate foo;
+    """, """
+        extern crate foo;
+    """)
+
+    fun `test change extern crate visibility`() = doTestChanged("""
+        extern crate foo;
+    """, """
+        pub extern crate foo;
+    """)
+
+    fun `test change mod item to mod decl`() = doTestChanged("""
+        mod foo {}
+    """, """
+        mod foo;
+    """)
+
+    fun `test change mod decl to mod item`() = doTestChanged("""
```

Comment:
Hmm, looks like we often do suck a reverse test. Idea: we can teach `doTestChanged` to do a reverse test automatically! e.g.
```
InlineFile(before)
doTest(replaceFileContent(after), shouldChange)
doTest(replaceFileContent(before), shouldChange)
```